diff --git a/src/server/game/Battlegrounds/Battleground.cpp b/src/server/game/Battlegrounds/Battleground.cpp
index 1c2d5fe9..783b067b 100644
--- a/src/server/game/Battlegrounds/Battleground.cpp
+++ b/src/server/game/Battlegrounds/Battleground.cpp
@@ -352,20 +352,17 @@ void Battleground::UpdateDumpenning(uint32 diff)
 
         ++m_DampeningCounter;
 
-        for (auto&& team : m_arenaTeams)
+        for (BattlegroundPlayerMap::const_iterator itr = m_Players.begin(); itr != m_Players.end(); ++itr)
         {
-            for (auto&& it : *team)
+            Player* player = ObjectAccessor::FindPlayer(itr->first);
+            if (player && player->GetMap() == GetBgMap())
             {
-                Player* player = ObjectAccessor::FindPlayer(it->Guid);
-                if (player && player->GetMap() == GetBgMap())
-                {
-                    ApplyDampening(player, m_DampeningCounter);
-                    if (Pet* pet = player->GetPet())
-                        ApplyDampening(pet, m_DampeningCounter);
-                    for (auto&& summon : player->GetSummons())
-                        if (summon->IsGuardian())
-                            ApplyDampening(summon, m_DampeningCounter);
-                }
+                ApplyDampening(player, m_DampeningCounter);
+                if (Pet* pet = player->GetPet())
+                    ApplyDampening(pet, m_DampeningCounter);
+                for (auto&& summon : player->GetSummons())
+                    if (summon->IsGuardian())
+                        ApplyDampening(summon, m_DampeningCounter);
             }
         }
     }
